flow/layers/physical_model_layer.h

```diff
@@ -16,9 +16,20 @@ class PhysicalModelLayer : public ContainerLayer {
   PhysicalModelLayer();
   ~PhysicalModelLayer() override;
 
-  void set_shape(std::unique_ptr<PhysicalLayerShape> shape) {
-    shape_ = std::move(shape);
+  void set_path(const SkPath& path) {
+    path_ = path;
+#if defined(OS_FUCHSIA)
+    SkRRect rrect = SkRRect::MakeEmpty();
+    if (path.isRRect(&rrect)) {
+      frameRRect_ = rrect;
+      isRect_ = rrect.isRect();
```

Comment:
I think we'll want the isRect check on all platforms so Paint() can avoid the canvas.saveLayer() call in the common case of a rectangular shape

Reply:
good catch